src/config.ts

```diff
@@ -159,14 +162,20 @@ export function defaultMeasurement(url: LocalUrl|RemoteUrl): Measurement {
  * Validate the given JSON object parsed from a config file, and expand it into
  * a fully specified configuration.
  */
-export async function parseConfigFile(parsedJson: unknown): Promise<Config> {
+export async function parseConfigFile(
+    parsedJson: unknown, configFilename?: string): Promise<Config> {
   const schema = require('./config.schema.json');
   const result =
       jsonschema.validate(parsedJson, schema, {propertyName: 'config'});
   if (result.errors.length > 0) {
     throw new Error(result.errors[0].toString());
   }
   const validated = parsedJson as ConfigFile;
+  if (!('$schema' in validated) && configFilename) {
+    // Extra IDE features can be activated if the config file has a schema.
+    validated.$schema = 'https://unpkg.com/tachometer/lib/config.schema.json';
```

Comment:
Somehow I feel that the schema should be at the top of the file, so in that case I think we'd need:

```
validated = {
  $schema: '...',
  ...validated,
};
```

Reply:
+1, done